MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/DottedAndSolidLineIconActivity.java

```diff
@@ -0,0 +1,166 @@
+          @Override
+          public void onStyleLoaded(@NonNull Style style) {
+
+            // Add the plane SymbolLayer icon to the map
+            style.addImage("plane-icon-id", BitmapFactory.decodeResource(
+              getResources(), R.drawable.ic_action_plane));
+            style.addSource(new GeoJsonSource("plane-source-id", MINNEAPOLIS_CITY_AIRPORT_POINT));
+            style.addLayer(new SymbolLayer("plane-symbol-layer-id", "plane-source-id").withProperties(
+              iconImage("plane-icon-id"),
+              iconSize(1.2f),
+              iconIgnorePlacement(true),
+              iconAllowOverlap(true)
+            ));
+
+            // Add the marker icons which represent the plane's origin and destination locations
+            style.addImage("destination-and-origin-icon-id", BitmapFactory.decodeResource(
+              getResources(), R.drawable.red_marker));
+            style.addSource(new GeoJsonSource("destination-and-origin-source-id",
+              FeatureCollection.fromFeatures(
+                new Feature[] {
+                  Feature.fromGeometry(NEWARK_AIRPORT_POINT),
+                  Feature.fromGeometry(SAN_FRANCISCO_AIRPORT_POINT)
+                })));
+            style.addLayer(new SymbolLayer("destination-and-origin-symbol-layer-id",
+              "destination-and-origin-source-id").withProperties(
+              iconImage("destination-and-origin-icon-id"),
+              iconSize(1f),
+              iconIgnorePlacement(true),
+              iconOffset(new Float[] {0f, -8f}),
+              iconAllowOverlap(true)
+            ));
+
+            // Add the dotted line LineLayer and use runtime-styling to style it
+            List<Point> dotList = new ArrayList<>();
+            dotList.add(NEWARK_AIRPORT_POINT);
+            dotList.add(MINNEAPOLIS_CITY_AIRPORT_POINT);
+            style.addSource(new GeoJsonSource("dotted-line-source-id",
+              Feature.fromGeometry(LineString.fromLngLats(dotList))));
+            style.addLayerBelow(
+              new LineLayer("dotted-line-layer-id", "dotted-line-source-id")
+                .withProperties(lineWidth(4.5f),
+                  lineColor(Color.RED),
+                  lineDasharray(new Float[] {1f, 1f})), "plane-symbol-layer-id");
+
+
+            // Add the solid line LineLayer and use runtime-styling to style it
+            List<Point> solidLineList = new ArrayList<>();
+            solidLineList.add(MINNEAPOLIS_CITY_AIRPORT_POINT);
+            solidLineList.add(SAN_FRANCISCO_AIRPORT_POINT);
+            style.addSource(new GeoJsonSource("solid-line-source-id",
+              Feature.fromGeometry(LineString.fromLngLats(solidLineList))));
+            style.addLayerBelow(new LineLayer("solid-line-layer-id", "solid-line-source-id")
+              .withProperties(lineWidth(4.5f),
+                lineColor(Color.BLUE)), "plane-symbol-layer-id");
+          }
+        });
```

Comment:
we are creating 3 source objects can we merge them to be one? For points vs lines nothing needs to be done but for solid/dot list we will need to add some kind of filter.